### DISCUSSION TITLE
When `query` is undefined useSWR will fail silently (upgrade 1.30->2.0.4)

### DISCUSSION THREAD

**Discussion by sandrodz** [vercel/swr#2513] [Q&A]
# Bug report

## Description / Observed Behavior

When `query` is undefined useSWR will fail silently and will not trigger any error.
in v 1.3.0 undefined wouldn't casue this issue.

```
import useSWR from 'swr';

import ApiService from '@/services/Api.service';

const useApi = (endpoint, query) => {
  const { data, error } = useSWR([endpoint, query], ApiService);

  return {
    loading: !data && !error,
    data: data?.data,
    error,
  };
};

export default useApi;
```

## Expected Behavior

Request should succeed.

## Additional Context

SWR update from 1.3.0 to 2.0.4

**Comment by promer94** [comment]
I can't reproduce your problem https://codesandbox.io/s/2446-forked-e4nbnn?file=/src/App.js

Fetcher no longer accepts multiple arguments in v2.  Maybe this is the cause of your problem ?
https://swr.vercel.app/blog/swr-v2#fetcher-no-longer-accepts-multiple-arguments
